core/LSU.hpp

```diff
@@ -49,6 +49,14 @@ namespace olympia
 
             // Parameters for ldst_inst_queue
             PARAMETER(uint32_t, ldst_inst_queue_size, 8, "LSU ldst inst queue size")
+            PARAMETER(uint32_t, replay_buffer_size, ldst_inst_queue_size, "Replay buffer size")
+            PARAMETER(uint32_t, replay_issue_delay, 3, "Replay Issue delay")
+            // LSU microarchitecture parameters
+            PARAMETER(bool, allow_speculative_load_exec, true, "Allow loads to proceed speculatively before all older store addresses are known")
+            // Pipeline length
+            PARAMETER(uint32_t, mmu_lookup_stage_length, 1, "Length of the mmu lookup stage")
+            PARAMETER(uint32_t, cache_lookup_stage_length, 1, "Length of the cache lookup stage")
+            PARAMETER(uint32_t, cache_read_stage_length, 1, "Length of the cache read stage")
```

Comment:
I designed the parameters to be in setup in a ldst_pipeline agnostic way
By default these are the mappings

| Stage         | Pipeline Alias     | Default |
|--------------|-----------|----------- |
| ADDRESS_GENERATION | 0      | 0 |
| MMU_LOOKUP      |  mmu_lookup_stage | 1 |
| CACHE_LOOKUP      |  mmu_lookup_stage + cache_lookup_stage | 2 |
| CACHE_READ      |  mmu_lookup_stage + cache_lookup_stage + cache_read_stage | 3 |
| COMPLETE      |  mmu_lookup_stage + cache_lookup_stage + cache_read_stage + 1 | 4 |

This would ensure that the stages are in the order without checking for the overlaps as the arithmetic would handle it.I should add validation to ensure that the length parameters are atleast 1.

example:
```
mmu_lookup_stage_length => 1
cache_lookup_stage_length => 2
cache_read_stage_length =>  2
```
| Stage         | Pipeline Alias     | Value |
|--------------|-----------| -----------|
| ADDRESS_GENERATION | 0      | 0  |
| MMU_LOOKUP      |  mmu_lookup_stage | 1|
| CACHE_LOOKUP      |  mmu_lookup_stage + cache_lookup_stage | 3|
| CACHE_READ      |  mmu_lookup_stage + cache_lookup_stage + cache_read_stage | 5 |
| COMPLETE      |  mmu_lookup_stage + cache_lookup_stage + cache_read_stage + 1 | 6 |

Reply:
Ok, I can buy that for a dollar 💵 (sorry, bad movie reference).